README.md

```diff
@@ -26,7 +26,7 @@ as [Kaniko](https://github.com/GoogleContainerTools/kaniko),
 | Dependency                                | Supported versions                     |
 | ----------------------------------------- | -------------------------------------- |
 | [Kubernetes](https://kubernetes.io/)      | v1.15.\*, v1.16.\*, v1.17.\*, v1.18.\* |
-| [Tekton](https://cloud.google.com/tekton) | v0.20.1                                |
```

Comment:
Can we describe the _minimum_ version of Tekton that's required by Shipwright? This makes it seem like 0.21 is the minimum, which might make it harder for people to try out.

Same with minimum Kubernetes versions above; saying 1.15 - 1.18 makes it sound like 1.19+ _aren't_ supported for some reason.

Reply:
Thanks @ImJasonH !
Currently, in our side, we don't have a flow to test the minimum version of tekton for each shipwright-io build release. At present, for each build release, we only bump tekton to the version what we want and test it. 

So, to some extent, for a fixed build release, users need to have a fixed tekton version. But yes, this is a good suggestion. Maybe we can add a flow to test the minimum tekton version for each build release in the future, then we can document it.

Reply:
I understand we might not have tests for every Tekton version, but I think if we believe it works with 0.20 or 0.19 we should document that. If lacking tests is a concern we can invest in that as well. GitHub Actions can run against a matrix of Tekton versions relatively easily if we feel that's necessary to test.

Shipwright should be expected to work with a range of Tekton versions, or else there's no way to upgrade either Tekton or Shipwright safely, since they'd have to be done in lock-step at the same time. Users who have an older Tekton version (even just the previous version) might be scared off to learn they need the absolute latest Tekton release to use Shipwright.

This feedback doesn't need to block this PR, but we should definitely not be documenting that Shipwright only works with one version of Tekton.

Reply:
@ImJasonH we tried the matrix before for the k8s version permutations. I think github actions started to behave strange. I need to revisit what was exactly the problem. All I´m trying to say is that infrastructure wise, github might not be ideal for so many flavors of testing, we need to revisit this again. 

I see the value on not scaring users in regards of Tekton versions. So we definitely need to do better in here.

I will open a new issue in our backlog that covers:

- How to properly document minimum version for k8s and tekton, for Shipwright users in order to avoid situations as the one you jsut mentioned on `might be scared`.
- Provide a follow-up on how can we have a test that can continuously test N,N-1,N-2 versions of Tekton and k8s. Plus a layout of pros and cons based on what we found when @SaschaSchwarze0 tried it.